### PR TITLE
timings/plots

### PR DESCRIPTION
**Pull request by PhilipDeegan**
Mostly additions of a new diagnostic to log the particle counts per patch

timing things were moved to phlop as much as possible

## Summary by CodeRabbit

- **New Features**
	- Introduced a logging system to set log levels and retrieve logs.
	- Added `InfoDiagnostics` for enhanced diagnostic data handling.
	- Improved error handling and default parameter settings in `PhareSeeRunner`.
	- Enhanced logging with log levels across various modules for better data categorization and error tracing.
	- New functionality in managing dependencies and project configurations using CMake.
	- Added new classes and methods for timing execution and reporting in simulations.
	- Introduced new plotting tools for visualizing run timing data.

- **Enhancements**
	- Updated MPI library handling and detection logic.
	- Enhanced logging scopes with specific log levels for better granularity.
	- Added new diagnostic types and improved management in `DiagnosticsManager`.

- **Bug Fixes**
	- Corrected a typo in a comment regarding particle populations.

### CODE REVIEW

**Comment by PhilipDeegan** [comment]
macos build failing should be resolved by https://github.com/PHAREHUB/PHARE/pull/829

File: pyphare/pyphare/pharesee/run.py [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 import numpy as np
-from pyphare.pharesee.hierarchy import compute_hier_from
-from pyphare.pharesee.hierarchy import compute_hier_from_
 
-from .hierarchy import flat_finest_field, hierarchy_from
+from .hierarchy import (
+    compute_hier_from,
+    flat_finest_field,
+    hierarchy_from,
+)
```
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply pyphare/pyphare/pharesee/run.py:8]: #doubt

File: src/restarts/restarts_manager.hpp [hunk 2]
```diff
@@ -154,7 +154,10 @@ void RestartsManager<Writer>::dump(double timeStamp, double timeStep)
         return; // not active
 
     if (needsWrite_(*restarts_properties_, timeStamp, timeStep))
+    {
+        PHARE_LOG_SCOPE(3, "RestartsManager::dump");
```
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply src/restarts/restarts_manager.hpp:158]: there is another option to use string concatenation via the macro system

eg: https://github.com/mkn/mkn.kul/blob/master/inc/mkn/kul/log.hpp#L253-L256

what this ends up looking like is 

```
KLOG(INF) << "info logging line";
KLOG(ERR) << "error logging line";
KLOG(DBG) << "debug logging line";
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required (VERSION 3.20.1) # released April 8, 2021 - https://www.kitware.com/cmake-3-20-1-available-for-download/
 
-project(PHARE VERSION 0.1 LANGUAGES CXX)
+project(PHARE VERSION 0.1 LANGUAGES CXX C)
```
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply CMakeLists.txt:3]: I have a bunch of quality of life changes in this PR that I should probably isolate, 

this is for MPI, we only use C MPI, and to tell `FindPackage` we only want the C API, we have to include it here

File: pyphare/pyphare/pharesee/run.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply pyphare/pyphare/pharesee/run.py:8]: explain why you think there is a cyclic import

File: tools/python3/phloping.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            final_time + ts,
+            final_time / (final_time / time_step) / sub_steps,
+        )
+
+    def steps_per_coarse_timestep_for_L(self, i):
+        return substeps_per_finer_level**i
+
+    def normalised_times_for_L(self, ilvl):
+        """
+        Normalise substep time against particular count for that level
```
Comment (PhilipDeegan) [inline tools/python3/phloping.py:125]: typo

File: src/core/utilities/logger/logger_defaults.hpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+#define PHARE_CORE_UTILITIES_LOGGER_LOGGER_DEFAULTS_HPP
+
+#include "core/def/phlop.hpp"
+
+#if PHARE_HAVE_PHLOP
+#define PHARE_SCOPE_TIMER PHLOP_SCOPE_TIMER
+#endif // PHARE_WITH_PHLOP
+
+#ifndef PHARE_SCOPE_TIMER
+#define PHARE_SCOPE_TIMER // nothing
```
Comment (PhilipDeegan) [inline src/core/utilities/logger/logger_defaults.hpp:11]: confirm this doesn't need to be a function

File: pyphare/pyphare/pharesee/run.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply pyphare/pyphare/pharesee/run.py:8]: so there is no cyclic import?

**Review by nicolasaunai — APPROVED** [review]

**Pull request merged**